client/state/comments/reducer.js

```diff
@@ -364,23 +365,107 @@ export const activeReplies = createReducer(
 	}
 );
 
-export const counts = ( state = {}, action ) => {
-	const { type, ...actionData } = action;
-	if ( COMMENT_COUNTS_UPDATE === type ) {
-		const { siteId, postId, ...commentCounts } = actionData;
-		if ( ! siteId ) {
-			return state;
-		}
-		const keyName = siteId && postId ? postId : 'site';
-		const siteCounts = {
-			[ siteId ]: Object.assign( {}, state[ siteId ] || {}, {
-				[ keyName ]: commentCounts,
-			} ),
-		};
-		return Object.assign( {}, state, siteCounts );
+function updateCount( counts, rawStatus, value = 1 ) {
+	const status = rawStatus === 'unapproved' ? 'pending' : rawStatus;
+	if ( ! counts || ! counts[ status ] ) {
```

Comment:
From the quick look at the code, I think the second part of this condition is causing an incorrect bail when the count for given status is `0`.

Reply:
Thanks updated in 6312b19